### PR TITLE
Monitor Revamp

### PR DESCRIPTION
**Pull request by dhaavi**

### CODE REVIEW

File: modules/monitor/src/components/Monitor.vue [hunk 1]
```diff
@@ -107,100 +120,109 @@
...
+        <span v-if="selectedScope.name == 'IH'">Incoming from Localhost</span>
+        <span v-else-if="selectedScope.name == 'IL'">Incoming from the LAN</span>
+        <span v-else-if="selectedScope.name == 'II'">Incoming from the Internet</span>
+        <span v-else-if="selectedScope.name == 'IX'">Incoming - Invalid</span>
+        <span v-else-if="selectedScope.name == 'PH'">Peers on Localhost</span>
+        <span v-else-if="selectedScope.name == 'PL'">Peers on the LAN</span>
+        <span v-else-if="selectedScope.name == 'PI'">Peers on the Internet</span>
+        <span v-else-if="selectedScope.name == 'PX'">Peers - Invalid</span>
+        <span v-else>{{ selectedScope.name }}</span>
       </h2>
```
Comment (ppacher) [inline modules/monitor/src/components/Monitor.vue:150]: That code is duplicated, isn't it possible to render a `<template>` at multiple places? (disclaimer, I don't really know Vue yet)

File: modules/monitor/src/components/Monitor.vue [hunk 2]
```diff
@@ -107,100 +120,109 @@
...
-                <td>{{ selectedCommunication.Intel }}</td>
-              </tr> -->
               <tr>
-                <td>Active</td>
-                <td v-if="!selectedCommunication.FirstLinkEstablished && !selectedCommunication.LastLinkEstablished">no activity</td>
-                <td v-else-if="selectedCommunication.FirstLinkEstablished == selectedCommunication.LastLinkEstablished">at {{ selectedCommunication.FirstLinkEstablished|fmt_time }}</td>
-                <td v-else>{{ selectedCommunication.FirstLinkEstablished|fmt_time }} - {{ selectedCommunication.LastLinkEstablished|fmt_time }}</td>
+                <td>Entity</td>
+                <td>
+                  <PRE>{{ selectedScope.dnsRequest.Entity }}</PRE>
```
Comment (ppacher) [inline modules/monitor/src/components/Monitor.vue:179]:
```suggestion
                  <pre>{{ selectedScope.dnsRequest.Entity }}</pre>
```

File: modules/monitor/src/components/Monitor.vue [hunk 3]
```diff
@@ -245,160 +265,242 @@ export default {
...
   computed: {
-    anyActivity() {
-      for (var i = 0; i < this.tree.length; i++) {
-        if (this.tree[i]._children && this.tree[i]._children.length > 0) {
-          return true;
-        }
-      }
-      return false;
-    },
     tree() {
```
Comment (ppacher) [inline modules/monitor/src/components/Monitor.vue:318]: This function is very very long, can we split it up?

Comment (dhaavi) [reply modules/monitor/src/components/Monitor.vue:318]: Will have to look into this later, fast fixes would only make it minimally better.

**Pull request merged**